src/detect-engine-threshold.c

```diff
@@ -69,6 +69,11 @@
 #include "util-var-name.h"
 #include "tm-threads.h"
 
+#define TIME_DIFF_SEC(p_ts, lookup_tsh) \
```

Comment:
With renamed params, this seems like it'd be a good add to `util-time.h`

Reply:
Would you like me to move it as part of this PR?

Reply:
I think it'd be a good addition so yes.

Reply:
Done in #4747. This has a knock on effect where it made sense to change the threshold structure to hold a timeval (tv1) instead of holding tv_sec1 and tv_usec1 separately. I think the consistency is better.